### ISSUE TITLE
NoNodesAvailableException occurs in testing environment when model is deleted

### ISSUE THREAD

**Issue by jdssem** [babenkoivan/elastic-scout-driver-plus#115]
| Software                   | Version
| --------------------| ---------------
| PHP                           | 8.0.5
| Elasticsearch            | 7.10
| Laravel                      | 8.81.0
| Laravel Scout           | 8.6.1
| Elastic Scout Driver | 2.0.0

**Describe the bug**
After upgrading to v3.2.3, `NoNodesAvailableException` occurs if a model is deleted(or forceDeleted) in my test even if I set `SCOUT_DRIVER` to null. It works perfectly without exception on v3.2.2.

Stack trace
```
Elasticsearch\Common\Exceptions\NoNodesAvailableException: No alive nodes found in your cluster

/Project/vendor/elasticsearch/elasticsearch/src/Elasticsearch/ConnectionPool/StaticNoPingConnectionPool.php:64
/Project/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Transport.php:82
/Project/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Transport.php:99
/Project/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php:297
/Project/vendor/react/promise/src/FulfilledPromise.php:28
/Project/vendor/ezimuel/ringphp/src/Future/CompletedFutureValue.php:55
/Project/vendor/ezimuel/ringphp/src/Core.php:341
/Project/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php:345
/Project/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php:241
/Project/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Transport.php:110
/Project/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Client.php:1929
/Project/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Client.php:347
/Project/vendor/babenkoivan/elastic-adapter/src/Documents/DocumentManager.php:83
/Project/vendor/babenkoivan/elastic-scout-driver-plus/src/Jobs/RemoveFromSearch.php:43
/Project/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:36
/Project/vendor/laravel/framework/src/Illuminate/Container/Util.php:40
/Project/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:93
/Project/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:37
/Project/vendor/laravel/framework/src/Illuminate/Container/Container.php:653
/Project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php:128
/Project/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:128
/Project/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:103
/Project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php:132
/Project/app/Actions/BusDispatcher.php:12
/Project/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php:120
/Project/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:128
/Project/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:103
/Project/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php:122
/Project/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php:70
/Project/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php:98
/Project/vendor/laravel/framework/src/Illuminate/Queue/SyncQueue.php:43
/Project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php:253
/Project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php:229
/Project/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php:77
/Project/vendor/laravel/framework/src/Illuminate/Foundation/Bus/PendingDispatch.php:193
/Project/vendor/babenkoivan/elastic-scout-driver-plus/src/Searchable.php:57
/Project/vendor/laravel/scout/src/Searchable.php:46
/Project/vendor/laravel/framework/src/Illuminate/Macroable/Traits/Macroable.php:124
/Project/vendor/laravel/scout/src/Searchable.php:176
/Project/vendor/laravel/scout/src/ModelObserver.php:121
/Project/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php:424
/Project/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php:249
/Project/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasEvents.php:189
/Project/vendor/laravel/framework/src/Illuminate/Database/Eloquent/SoftDeletes.php:54
/Project/vendor/laravel/framework/src/Illuminate/Support/helpers.php:263
/Project/vendor/laravel/framework/src/Illuminate/Database/Eloquent/SoftDeletes.php:56
/Project/tests/Feature/MyTest.php:136
```

**To Reproduce**

phpunit.xml
```
<env name="SCOUT_DRIVER" value="null"/>
```

make a test that some model is deleted
```
$model->delete()
```

**Current behavior**

throwing `NoNodesAvailableException`

**Expected behavior**

Scout driver should not be enabled in testing environment.
(`RemoveFromSearch` should not be fired.)

**Comment by nicopenaredondo** [comment]
Have you tried using "collection" as the SCOUT_DRIVER ?

**Comment by jdssem** [comment]
> Have you tried using "collection" as the SCOUT_DRIVER ?

`collection` doesn't work either. But I solved this problem temporarily by changing `SCOUT_QUEUE=false`

**Comment by nicopenaredondo** [comment]
@jdssem  can you share your test cases ? I'm having a problem also.

**Comment by nicopenaredondo** [comment]
> > Have you tried using "collection" as the SCOUT_DRIVER ?
> 
> `collection` doesn't work either. But I solved this problem temporarily by changing `SCOUT_QUEUE=false`

I'm having this kind of error `Call to undefined method Laravel\\Scout\\Engines\\NullEngine::executeSearchRequest()` when using a null driver upon fetching an aggregated results

**Comment by babenkoivan** [comment]
@jdssem I've just released [v3.2.4](https://github.com/babenkoivan/elastic-scout-driver-plus/releases/tag/v3.2.4), which might solve your issue.

@nicopenaredondo @jdssem please note that `elastic-scout-driver-plus` is not compatible with other Scout drivers, because it has a different interface. You can use [elastic-scout-driver](https://github.com/babenkoivan/elastic-scout-driver) if you need a generic Scout solution, but it lacks many Elasticsearch specific features.

**Comment by nicopenaredondo** [comment]
@babenkoivan i'll check your latest release. Thanks!
